packages/date-picker/src/DatePicker/components/Calendar/Calendar.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { Dispatch, SetStateAction, useEffect, useRef } from "react"
+import DayPicker from "react-day-picker/DayPicker"
+import FocusLock from "react-focus-lock"
+import cx from "classnames"
+import { NavbarElementProps } from "react-day-picker/types/Props"
+import {
+  BeforeAfterModifier,
+  DayModifiers,
+  RangeModifier,
+} from "react-day-picker/types/Modifiers"
+import { daysToNumbers } from "../../../utils/daysToNumbers"
+import { defaultDatePickerClasses } from "../../DatePickerClasses"
+import datePickerStyles from "../../DatePicker.scss"
+import { DayOfWeek } from "../../DatePicker"
+import { CalendarNav } from "../CalendarNav"
+
+export type CalendarProps = {
+  setPopperElement: Dispatch<SetStateAction<HTMLDivElement | null>>
+  styles: { [key: string]: React.CSSProperties }
+  attributes: {
+    [key: string]:
+      | {
+          [key: string]: string
+        }
+      | undefined
+  }
+  classNameAndIHaveSpokenToDST?: string
+  value?: Date
+  initialMonth?: Date
+  firstDayOfWeek: number
+  disabledDates?: Date[]
+  disabledDaysOfWeek?: DayOfWeek[]
+  disabledRange?: RangeModifier
+  disabledBefore?: Date
+  disabledAfter?: Date
+  disabledBeforeAfter?: BeforeAfterModifier
+  handleOnDayChange: (day: Date, modifiers: DayModifiers) => void
+  handleKeyDown: (e: any) => void
+}
+
+export type CalendarNavProps = Pick<
+  NavbarElementProps,
+  "onPreviousClick" | "onNextClick"
+>
+
+export const Calendar: React.VFC<CalendarProps> = ({
+  setPopperElement,
+  styles,
+  attributes,
+  classNameAndIHaveSpokenToDST,
+  value,
+  initialMonth,
+  firstDayOfWeek,
+  disabledDates,
+  disabledDaysOfWeek,
+  disabledRange,
+  disabledBefore,
+  disabledAfter,
+  disabledBeforeAfter,
+  handleOnDayChange,
+  handleKeyDown,
+}) => {
+  const calendarRef = useRef<HTMLDivElement>(null)
+
+  const getNavbar = ({ ...navbarProps }: CalendarNavProps) => (
+    <CalendarNav {...navbarProps} />
+  )
+
+  useEffect(() => {
+    if (value) {
+      if (calendarRef && calendarRef.current) {
```

Comment:
Looks like `if (calendarRef && calendarRef.current)` is a condition for both the `if` and the `else` block, you should be able tp pull it out as a guard clause instead and remove the nestes if statement.